lib/s3-s3.js

```diff
@@ -40,9 +46,25 @@ var s3 = function (primaryS3, secondaryS3) {
           setupOn(realRequest, 'extractError', onFunc);
           setupOn(realRequest, 'extractData', onFunc);
           setupOn(realRequest, 'success', onFunc);
-          setupOn(realRequest, 'error', onFunc);
           setupOn(realRequest, 'complete', onFunc);
 
+          // error is a special on() call which we use to see if this is a call that needs to
+          // go to the secondary
+          realRequest.on('error', function(error) {
+            if (usingPrimary && error &&
+                error.hasOwnProperty('code') &&
+              (error.code === 400 || error.code >= 500)) {
```

Comment:
I'm more concerned that a normal 500 or other case just doesn't hit the "error.code" way I've done things, since the error object is so poorly documented (aka not documented).